lib/memory_profiler/cli.rb

```diff
@@ -4,21 +4,42 @@
 
 module MemoryProfiler
   class CLI
+    BIN_NAME     = "ruby-memory-profiler"
+    VERSION_INFO = "#{BIN_NAME}  #{MemoryProfiler::VERSION}"
+
     STATUS_SUCCESS = 0
     STATUS_ERROR   = 1
 
+    DEFAULTS = {
+      ignore_files: "memory_profiler/lib"
+    }.freeze
+
+    REPORTER_KEYS = [
+      :top, :trace, :ignore_files, :allow_files
+    ].freeze
+
+    RESULTS_KEYS = [
+      :to_file, :color_output, :retained_strings, :allocated_strings,
+      :detailed_report, :scale_bytes, :normalize_paths
+    ].freeze
+
+    private_constant :BIN_NAME, :VERSION_INFO,:STATUS_SUCCESS, :STATUS_ERROR,
+                     :DEFAULTS, :REPORTER_KEYS, :RESULTS_KEYS
+
+    #
+
```

Comment:
Overall - 👍  
Messages, code and option names looks cleaner. 👍 for `--pretty`

Not sure about the need for this 3 constants - I would just extract them into local variables (out of the loop). But let's delegate the final decision to Sam.

Reply:
Thanks for the feedback @fatkodima 
Cheers :)